File: CSES/IntroductoryProblems/IncreasingArray/increasingArray.cpp
#include <bits/stdc++.h>

int main()
{
    int n;
    std::cin >> n;

    std::vector<int> v;
    for (int i=0; i<n; i++)
    {
        int x;
        std::cin >> x;
        v.push_back(x);
    }
    // for (std::vector<int>::iterator it=v.begin(); it!=v.end(); it++)
    // {
    //     std::cout << *it << ", ";
    // }
    
    long long count{0};
    for (std::vector<int>::iterator it=v.begin(); it!=v.end(); it++)
    {
        int curr;
        int prev;
        if (it == v.begin())
        {
            continue;
        }
        else
        {
            curr = *it;
            prev = *(std::prev(it));
            // std::cout << "curr = " << curr << ", prev = " << prev << ", count = " << count << std::endl;
            if (curr < prev)
            {
                count += prev - curr;
                *it += prev - curr;
            }
            // std::cout << "curr = " << curr << ", prev = " << prev << ", count = " << count << std::endl;
        }
    }
    std::cout << count;

    return 0;
}

File: CSES/IntroductoryProblems/IncreasingArray/increasingArray.py
def main():
    n = int(input())
    line = input()
    v = []
    for el in line.split():
        v.append(int(el))
    count = 0
    for i in range(len(v)):
        if i == 0:
            continue
        else:
            curr = v[i]
            prev = v[i-1]
            if (curr < prev):
                count += prev - curr
                v[i] += prev - curr
    print (count)

if (__name__ == "__main__"):
    main()
